asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+
+# %% [markdown]
+# ### RQ 6.07 | What challenges have survey participants found with working in repairs, servicing and maintenance?
+# - SQ: 45
+# - Sample:
+# - Standard_output
+# <p>
+# N.B. Select all that apply type question
+
+# %%
+# Print possible answers and check counts
+data[col.q45[0]].value_counts()
+
+# %%
+def transform_45(x):
+    """
+    Function to be applied to dataframe which separates answers selected
+    by creating an individual tally column for each possible answer. 
+
+    Args:
+        x: pandas dataframe containing responses from analytical sample.
+    """
+
+    # Tally occurrence of each answer
+    if response in x[col.q45[0]]:
+        return response
+    else:
+        return None
+    
+# Specifying desired order for outputs
+ans_45 = ["Difficult to find the skills you need",
+"Low levels of customer demand",
+"Hard to make a profit from this work",
+"Difficult to fit this work into your work schedule",
+"No challenges", # re-coded new option
+"Poor original installations", # re-coded new option
+"Not enough manufacturer support", # re-coded new option
+"Difficulty sourcing components", # re-coded new option
+"Other",
+"Don't know",
+"Total"
+]
+
+# Create separate dataframe that stacks all answer columns
+df_45 = pd.DataFrame()
+for n in range(0, len(ans_45)-1):
+
+    # Create new column for each answer option
+    response = ans_45[n]
+    new_col = "SQ45Ans"+str(n+1)
+    data[new_col] = data.apply(transform_45, axis=1)
+
+    # Stack columns
+    df = data.filter(["EmploymentType", new_col])
+    df = df.rename(columns={new_col:"SQ45"})
+    df_45 = pd.concat([df_45, df])
+
+# Generate crosstab dataframe
+df_607_crosstab = crosstable("EmploymentType",
+                             df_45,
+                             "SQ45",
+                             ans_45
+                             )
+df_607_crosstab = df_607_crosstab.fillna(0)
+df_607_crosstab = df_607_crosstab.astype(int)
+
+# Remove Employee index row because they are not asked
+df_607_crosstab = df_607_crosstab.drop(index='Employee')
+
+# Save to .csv and display
+df_607_crosstab.to_csv("../../outputs/section6/csv/recoded/607_recoded.csv")
+df_607_crosstab
+
+# %%
+# Generate figure
+groupedbar(df_607_crosstab,
+           "607_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %%
+# Generate figure
+donut(df_607_crosstab,
+           "607_recoded",
+           "SQ 45. Have you found any challenges working in heat pump repairs, servicing and maintenance?")
+
+# %% [markdown]
+# ### RQ 6.08 | How do the challenges that survey participants find with working in repairs, servicing and maintenance differ depending on the size of the company they own?
+# - SQ: 45
+# - Sample:
+# - Company_size_owner
+
+# %%
+# CompanySizeOwner V2
+# Create separate dataframe that stacks all answer columns
```

Comment:
What do you mean here by stacks all answer column here?

Reply:
Apologies for my poor wording - I am not sure how to articulate this:

![image](https://github.com/nestauk/asf_installer_survey/assets/101257595/d98a44fa-588c-411b-bacc-34d262410627)


Please do let me know if you have any suggestions on a better way of describing this transformation :)

Reply:
Ah that makes sense! Something like "Transforms a DataFrame from a wide format with separate columns for different survey answers into a long format where each type of employment is listed with all associated answers in a single column."